### PR TITLE
Upgrading font-awesome from 4.7.0 to 5.0.10

### PR DESCRIPTION
**Pull request by gomesp**
The new font-awesome package uses the SVG with JS icon set. The previous
icon set has now all been renamed to the new icons present in FA5.

This change has been applied to the angular codebase only.

- Please make sure the below checklist is followed for Pull Requests.

- [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [x] Tests are added where necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/angular/src/main/webapp/app/layouts/index.ts.ejs [hunk 1]
```diff
@@ -28,3 +28,41 @@ export * from './navbar/active-menu.directive';
 export * from './profiles/page-ribbon.component';
 export * from './profiles/profile.service';
 export * from './profiles/profile-info.model';
+// Imports all fontawesome core and solid icons
+
+import fontawesome from '@fortawesome/fontawesome';
```
Comment (gmarziou) [inline generators/client/templates/angular/src/main/webapp/app/layouts/index.ts.ejs]: I'm not sure layouts/index.ts is the best place for these global imports, personally I would have put them in vendor.ts as we do for CSS but I could be wrong.

**Comment by gmarziou** [comment]
Have you tested a prod build? This is where most issues occur usually.

**Comment by gmarziou** [comment]
Travis build fails on app using SASS, you must edit also vendor.scss.ejs

**Comment by gomesp** [comment]
@gmarziou Thanks for pointing the sass issue. I have tested the generator with the default settings at first (so no sass). Can I run the travils build in my own fork before submitting the PR?

File: generators/client/templates/angular/src/main/webapp/app/layouts/index.ts.ejs (hunk 1 above)
Comment (gomesp) [reply generators/client/templates/angular/src/main/webapp/app/layouts/index.ts.ejs]: That's exactly why I wanted to submit the PR soon, so I could get some input of the preferred place for the global imports. I'll move it vendor.ts. 👍

File: generators/client/templates/angular/package.json.ejs [hunk 2]
```diff
@@ -36,7 +36,8 @@
     "@ng-bootstrap/ng-bootstrap": "1.0.1",
     "bootstrap": "4.0.0",
     "core-js": "2.5.3",
-    "font-awesome": "4.7.0",
+    "@fortawesome/fontawesome": "^1.1.5",
+    "@fortawesome/fontawesome-free-solid": "^5.0.10",
```
[the remark this answers is not in the document]
Comment (gomesp) [reply generators/client/templates/angular/package.json.ejs]: Thanks, will do! 👍

**Comment by pascalgrimaud** [comment]
Yes, just link your fork to your own Travis

**Comment by gomesp** [comment]
Hi team, I fixed the issues with the saas version and applied the feedback given to me in this PR.

The travis tests are now passing in my for (thanks for pointing out how to run travis on it) and I have generated 2 apps with these changes locally, one with sass and the other without sass.

I hope everything is in good shape now to close the issue #7516.

**Comment by gmarziou** [comment]
Looks good to me, maybe you could squash your 2 commits into a single one.

File: generators/client/templates/angular/src/main/webapp/app/vendor.ts.ejs [hunk 3]
```diff
@@ -27,4 +27,45 @@ import 'rxjs/add/operator/map';
 import 'rxjs/add/operator/catch';
 import 'rxjs/add/observable/of';
 import 'rxjs/add/observable/throw';
+
+// Imports all fontawesome core and solid icons
```
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/vendor.ts.ejs]: Doesn't it support web fonts anymore?
Does the SVG + JS approach have a significant saving of bundle size?

Comment (gomesp) [reply generators/client/templates/angular/src/main/webapp/app/vendor.ts.ejs]: This is the recommended approach by the FA project. They still support web fonts as well. I guess this is a core project decision, which direction you want to take (if different than the recommended one). I don't have a comparison of the bundle size between the options, but I can figure out if you guys want.

Comment (PierreBesson) [reply generators/client/templates/angular/src/main/webapp/app/vendor.ts.ejs]: The "SVG font" approach seems to have a lot of advantages and it's been a hot topic of discussion, the consensus being that they are better than "Icon Fonts". Here are 3 links about it :

https://www.ianfeather.co.uk/ten-reasons-we-switched-from-an-icon-font-to-svg/
https://cloudfour.com/thinks/seriously-dont-use-icon-fonts/
https://blog.github.com/2016-02-22-delivering-octicons-with-svg/

Especially now that it's supported by FA and easy to setup in our webpack build, I think it's safe to move to this approach.

Comment (deepu105) [reply generators/client/templates/angular/src/main/webapp/app/vendor.ts.ejs]: yup, I'm aware of it, I myself use SVG fonts in my projects, however, I was concerned about the JS part of if you look at some other SVG font libs they don't use JS. But then its not a huge concern

Comment (deepu105) [reply generators/client/templates/angular/src/main/webapp/app/vendor.ts.ejs]: so lets do it

Comment (deepu105) [reply generators/client/templates/angular/src/main/webapp/app/vendor.ts.ejs]: Also seems they will release a Angular version soon https://fontawesome.com/get-started/advanced-options

Comment (deepu105) [reply generators/client/templates/angular/src/main/webapp/app/vendor.ts.ejs]: we could migrate to the angular version when released

**Comment by gomesp** [comment]
Woot, I got a PR merged! 🎉

**Comment by pascalgrimaud** [comment]
thanks and congrats @gomesp :-)

**Comment by jdubois** [comment]
Awesome, congrats!!

**Pull request merged**